File: samples/spring-mvc-boot-3-lombok/src/main/java/io/openapiprocessor/samples/ApiController.java
package io.openapiprocessor.samples;

import io.openapiprocessor.openapi.api.BarApi;
import io.openapiprocessor.openapi.api.FooApi;
import io.openapiprocessor.openapi.model.Foo;
import org.springframework.web.bind.annotation.RestController;

import java.util.UUID;

@RestController
public class ApiController implements FooApi, BarApi {

    @Override
    public Foo postFoo(Foo foo) {
        return foo != null ? foo : createFoo();
    }

    @Override
    public MappedBar postBar(MappedBar bar) {
        return bar != null ? bar : createBar();
    }

    private static Foo createFoo() {
        return Foo.builder()
                .id(UUID.randomUUID())
                .foo("foo")
                .bar(createBar())
                .build();
    }

    private static MappedBar createBar() {
        return MappedBar.builder()
                .bar("bar")
                .build();
    }

}

File: samples/micronaut-4-kt/src/main/kotlin/io/openapiprocessor/samples/ApiController.kt
package io.openapiprocessor.samples

import io.micronaut.http.annotation.Controller
import io.openapiprocessor.openapi.api.BarApi
import io.openapiprocessor.openapi.api.FooApi
import io.openapiprocessor.openapi.model.Bar
import io.openapiprocessor.openapi.model.Foo
import java.util.*

//@Controller
@Controller("/foo/bar/v1")
open class ApiController: FooApi, BarApi {

    override fun postFoo(foo: Foo?): Foo {
        return foo ?: Foo("foo", UUID.randomUUID(), Bar("value"))
    }

    override fun postBar(bar: Bar?): Bar {
        return bar ?: Bar("bar value")
    }
}
